app/src/main/java/org/thoughtcrime/securesms/onboarding/pickname/PickDisplayNameActivity.kt

```diff
@@ -0,0 +1,156 @@
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.launch
+import network.loki.messenger.R
+import org.session.libsession.utilities.AppTextSecurePreferences
+import org.thoughtcrime.securesms.ApplicationContext
+import org.thoughtcrime.securesms.BaseActionBarActivity
+import org.thoughtcrime.securesms.onboarding.messagenotifications.MessageNotificationsActivity
+import org.thoughtcrime.securesms.onboarding.messagenotifications.startMessageNotificationsActivity
+import org.thoughtcrime.securesms.ui.AppTheme
+import org.thoughtcrime.securesms.ui.PreviewTheme
+import org.thoughtcrime.securesms.ui.base
+import org.thoughtcrime.securesms.ui.baseBold
+import org.thoughtcrime.securesms.ui.components.OutlineButton
+import org.thoughtcrime.securesms.ui.contentDescription
+import org.thoughtcrime.securesms.ui.outlinedTextFieldColors
+import org.thoughtcrime.securesms.util.setUpActionBarSessionLogo
+import javax.inject.Inject
+
+private const val EXTRA_PICK_NEW_NAME = "extra_pick_new_name"
+
+@AndroidEntryPoint
+class PickDisplayNameActivity : BaseActionBarActivity() {
+
+    @Inject
+    lateinit var viewModelFactory: PickDisplayNameViewModel.AssistedFactory
+
+    private val viewModel: PickDisplayNameViewModel by viewModels {
+        val pickNewName = intent.getBooleanExtra(EXTRA_PICK_NEW_NAME, false)
+        viewModelFactory.create(pickNewName)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setUpActionBarSessionLogo()
+
+        ComposeView(this)
+            .apply { setContent { DisplayNameScreen(viewModel) } }
+            .let(::setContentView)
+
+        lifecycleScope.launch {
+            viewModel.eventFlow.collect {
+                startMessageNotificationsActivity()
+            }
+        }
+    }
+
+    @Composable
+    private fun DisplayNameScreen(viewModel: PickDisplayNameViewModel) {
+        val state = viewModel.stateFlow.collectAsState()
+
+        AppTheme {
+            DisplayName(state.value, viewModel::onChange) { viewModel.onContinue(this) }
+        }
+    }
+
+    @Preview
+    @Composable
+    fun PreviewDisplayName() {
+        PreviewTheme(R.style.Classic_Dark) {
+            DisplayName(State())
+        }
+    }
+
+    @Composable
+    fun DisplayName(state: State, onChange: (String) -> Unit = {}, onContinue: () -> Unit = {}) {
+        Column(
+            verticalArrangement = Arrangement.spacedBy(20.dp),
+            modifier = Modifier
+                .padding(horizontal = 50.dp)
+                .padding(bottom = 12.dp)
+        ) {
+            Spacer(modifier = Modifier.weight(1f))
+            Text(stringResource(state.title), style = MaterialTheme.typography.h4)
+            Text(
+                stringResource(state.description),
+                style = MaterialTheme.typography.base,
+                modifier = Modifier.padding(bottom = 12.dp))
+
+            OutlinedTextField(
```

Comment:
This does not apply the `base` text stye. If we are using a few of these we should create a SessionOutlineTextField, to be reused throughout.

Reply:
Also this textfield changes its width as you type more characters